File: is_subsequence_leetcode#392.rb
def is_subsequence(s, t)
    if s.length == 0
        return true
    end
    ele = s[0]
    idx = t.index(s[0])
    if idx == nil
        return false
    else
        return is_subsequence(s[1..-1], t[idx+1..-1])
    end
    
end

p is_subsequence('abc', 'ahbgdc')

File: Is_Subsequence_leetcode#392.js
var isSubsequence = function(s, t) {
    if(!s) return true;
    let ele = s[0];
    let idx = t.indexOf(ele);
    if(idx == -1){
        return false;
    } else {
        return isSubsequence(s.slice(1), t.slice(idx+1))
    }
};